### PR TITLE
Deprecate tierRank in NFT Reward Tier model

### PR DESCRIPTION
**Pull request by wraeth-eth**
## What does this PR do and why?

This was done as we can calculate this pretty easily where necessary. It is just an extra state that we need to manage. Every time an addition or deletion is performed, this needs to be recalculated.

## Screenshots or screen recordings

_If applicable, provide screenshots or screen recordings to demonstrate the changes._

## Acceptance checklist

- [ ] I have evaluated the [Approval Guidelines](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#approval-guidelines) for this PR.
- [ ] I have tested this PR in [all supported browsers](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#supported-browsers).
- [ ] I have tested this PR in dark mode and light mode (if applicable).

### CODE REVIEW

**Comment by wraeth-eth** [comment]
Current dependencies on/for this PR:
* main
  * **PR #2156** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/2156" target="_blank">Graphite</a> 
    * **PR #2169** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/2169" target="_blank">Graphite</a> 
      * **PR #2170** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/2170" target="_blank">Graphite</a> 
        * **PR #2171** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/2171" target="_blank">Graphite</a> 
          * **PR #2172** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/2172" target="_blank">Graphite</a> 
            * **PR #2181** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/2181" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/2181?utm_source=stack-comment).

File: src/components/NftRewards/NftPreview.tsx [hunk 1]
```diff
@@ -85,7 +87,7 @@ export function NftPreview({
       >
         <div style={headerRowStyle}>
           <h3 style={{ ...headerStyle, textTransform: 'uppercase' }}>
-            {projectMetadata?.name} – Tier {rewardTier.tierRank}
+            {projectMetadata?.name} – Tier {tierRank}
```
Comment (tomquirk) [inline src/components/NftRewards/NftPreview.tsx:90]: **suggestion** guess we should `Trans` this

File: src/models/nftRewardTier.ts [hunk 2]
```diff
@@ -7,7 +7,8 @@ export type NftRewardTier = {
   remainingSupply: number | undefined
   imageUrl: string // link to ipfs
   name: string
-  tierRank: number // cheapest tier is 1
+  /* @deprecated - now derived from comparing contributionFloor between each */
```
Comment (tomquirk) [inline src/models/nftRewardTier.ts:10]: Should we create a TODO issue for this? Once we launch the new Create flow, we'll be able to delete this right?

**Review by tomquirk — APPROVED** [review]

File: src/models/nftRewardTier.ts (hunk 2 above)
Comment (wraeth-eth) [reply src/models/nftRewardTier.ts:10]: yessir

**Comment by wraeth-eth** [comment]
**Graphite Merge Job**

Current status: ⏳ Queued to merge

This pull request is currently queued to be merged as part of a stack.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/2181?utm_source=stack-comment).


Job Reference: vtVBnYDD1aIW70DZJ1cT

**Pull request merged**